libraries/app/database_api.cpp

```diff
@@ -1308,34 +1320,31 @@ order_book database_api_impl::get_order_book( const string& base, const string&
    return result;
 }
 
-vector<market_volume> database_api::get_top_markets(uint32_t limit)const
+vector<market_ticker> database_api::get_top_markets(uint32_t limit)const
 {
    return my->get_top_markets(limit);
 }
 
-vector<market_volume> database_api_impl::get_top_markets(uint32_t limit)const
+vector<market_ticker> database_api_impl::get_top_markets(uint32_t limit)const
 {
    FC_ASSERT( _app_options && _app_options->has_market_history_plugin, "Market history plugin is not enabled." );
 
    FC_ASSERT( limit <= 100 );
 
    const auto& volume_idx = _db.get_index_type<graphene::market_history::market_ticker_index>().indices().get<by_volume>();
    auto itr = volume_idx.rbegin();
-   vector<market_volume> result;
+
+   vector<market_ticker> result;
    result.reserve(limit);
 
-   const fc::time_point_sec now = fc::time_point::now();
+   const fc::time_point_sec now = _db.head_block_time();
 
    while( itr != volume_idx.rend() && result.size() < limit)
    {
-      market_volume mv;
-      mv.time = now;
       const auto assets = get_assets( { itr->base, itr->quote } );
```

Comment:
I'd suggest that we use a small local cache in this `get_top_markets` function to reduce quantity of queries to the database.